Fix `declaration-block-no-redundant-longhand-properties`/`declaration-block-no-shorthand-property-overrides` false negatives for `font-variant`

Bilie opened the pull request:
> Which issue, if any, is this issue related to?

Closes https://github.com/stylelint/stylelint/issues/7620

> Is there anything in the PR that needs further explanation?

Hi, I have updated the properties file and added few tests, let me know if there is anything missing or can be improved

ybiquitous reviewed (changes requested):
@Bilie Thanks for the pull request! Can you address my review comments?

---

[suggest] Can you remove the following `skip` in `declaration-block-no-shorthand-property-overrides` test cases? It should pass now.

https://github.com/stylelint/stylelint/blob/b5b3799cc107ede5b418bef6ef64bab072b2deab/lib/rules/declaration-block-no-shorthand-property-overrides/__tests__/index.mjs#L211

ybiquitous commented on lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:
```diff
@@ -923,6 +923,77 @@ testRule({
+						font-variant: foo;
+					}
+				`,
+			message: messages.expected('font-variant'),
```
In addition, to test if a fixed code is in the correct order, can you replace only `foo` values with other values (e.g. `foo1`, `foo2`, etc.)?

Mouvedia commented on lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:
```diff
@@ -923,6 +923,77 @@ testRule({
+		{
+			code: stripIndent`
+					a {	
+						font-variant-ligatures: foo;
+						font-variant-position: foo;
+						font-variant-caps: foo;
+						font-variant-numeric: foo;
+						font-variant-alternates: foo;
+						font-variant-east-asian: foo;
+						font-variant-emoji: foo;
+						font-variant: foo;
```
medium

Could you put real values?
When all values are `foo` the test cannot detect that the order is incorrect from the `fixed` string.
This is partly my fault.

Bilie commented:
Hi thank you both for the review, it has been really helpful! 😊 I think I have addressed all comments, but if I have missed anything, just let me know

ybiquitous commented on lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:
```diff
@@ -68,6 +68,13 @@ testRule({
 			description: 'missing background-size',
 		},
+		{
+			code: stripIndent`
+					a {
+						font-variant: foo1 foo2 foo3 foo4 foo5 foo6 foo7;
+					}
+				`,
```
[suggest] Can you keep indentation consistent with other test cases?

```suggestion
			code: stripIndent`
				a {
					font-variant: foo1 foo2 foo3 foo4 foo5 foo6 foo7;
				}
			`,
```

ybiquitous commented on lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:
```diff
@@ -861,6 +868,25 @@ testRule({
 			description: 'CSS Text Decoration Module Level 4',
 		},
+		{
+			code: stripIndent`
+					a {
+						font-variant-ligatures: foo1;
+						font-variant-position: foo2;
+						font-variant-caps: foo3;
+						font-variant-numeric: foo4;
+						font-variant-alternates: foo5;
+						font-variant-east-asian: foo6;
+						font-variant-emoji: foo7;
+					}
+				`,
```
[suggest] Let's keep indentation consistent.

```suggestion
			code: stripIndent`
				a {
					font-variant-ligatures: foo1;
					font-variant-position: foo2;
					font-variant-caps: foo3;
					font-variant-numeric: foo4;
					font-variant-alternates: foo5;
					font-variant-east-asian: foo6;
					font-variant-emoji: foo7;
				}
			`,
```

ybiquitous commented on lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:
```diff
@@ -861,6 +868,25 @@ testRule({
+						font-variant-emoji: foo7;
+					}
+				`,
+			message: messages.expected('font-variant'),
+			fixed: stripIndent`
+				a {
+					font-variant: foo1 foo2 foo3 foo4 foo5 foo6 foo7;
+				}
+			`,
```
[suggest] Let's reorder `fixed` and `message` like other cases:

```suggestion
			fixed: stripIndent`
				a {
					font-variant: foo1 foo2 foo3 foo4 foo5 foo6 foo7;
				}
			`,
			message: messages.expected('font-variant'),
```

ybiquitous commented on lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:
```diff
@@ -923,6 +949,28 @@ testRule({
 
+testRule({
+	ruleName,
+	config: [true, { ignoreShorthands: ['/^font-variant/'] }],
```
[suggest] `ignoreShorthands` has been tested already, so can you remove this case?

Ref https://github.com/stylelint/stylelint/pull/7734/files#diff-1f86a8d8167e5474d3acb984827780d6f4fdc4b66d36478c75280f22c7e2b3cbR976

ybiquitous commented on lib/rules/declaration-block-no-shorthand-property-overrides/__tests__/index.mjs:
```diff
@@ -208,7 +208,6 @@ testRule({
 			message: messages.rejected('font-variant', 'font-variant-caps'),
 			line: 1,
 			endLine: 1,
```
[suggest] In addition, it'd be great if you could add `column`/`endColumn` like other cases 🙏🏼 

```suggestion
			line: 1,
			column: 36
			endLine: 1,
			endColumn: 48
```

ybiquitous reviewed (changes requested):
@Bilie Thanks! Can you take a look at my additional comments, please? This PR seems overall good, though.

Mouvedia commented:
off topic

It seems like there's a typo on https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant.
It should be `Specifies the keywords and functions related to the font-variant-ALTERNATES longhand property.`.

ybiquitous reviewed (approved):
Thank you. LGTM 👍🏼
